capstone-rs/src/capstone.rs

```diff
@@ -378,6 +380,25 @@ impl Capstone {
         }
     }
 
+    /// Returns `RegsAccess` structure for a given instruction
+    ///
+    /// Requires:
+    ///
+    /// 1. Instruction was created with detail enabled
+    /// 2. Skipdata is disabled
+    /// 3. Capstone was not compiled in diet mode
+    pub fn insn_regs_access<'s, 'i: 's>(&'s self, insn: &'i Insn) -> CsResult<InsnRegsAccess> {
```

Comment:
Instead of trying to do all of the error handling here, check the returned integer and convert that to a `CsResult`.
You can move that logic into 
That way, we don't have to manually track all possible errors that capstone should be handling anyway.

For examples, see `disasm()`. Also, `From<capstone_sys::cs_err::Type>` is implemented for `Error`, so you can use Error::from(capstone_sys_rc)` to do the error parsing for you.